sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -50,44 +56,98 @@ def load(cls, connection_string=None, endpoint=None, credential=None, **kwargs):
 
         key_vault_options = kwargs.pop("key_vault_options", None)
 
-        provider.__buildprovider(connection_string, endpoint, credential, key_vault_options)
+        if connection_string and endpoints:
+            raise AttributeError("Both connection_string and endpoints are set. Only one of these should be set.")
+        if endpoint and endpoints:
+            raise AttributeError("Both endpoint and endpoints are set. Only one of these should be set.")
+
+        # TODO: Currently Connection Strings are not supported for Geo Replication
+        if endpoints:
+            for endpoint_geo in endpoints:
+                provider._clients.append([endpoint_geo, provider.__build_provider(None, endpoint_geo, credential,
+                    key_vault_options)])
+        else:
+            store_endpoint = endpoint if endpoint is not None else provider.__parse_connection_string(connection_string)
+            provider._clients.append([store_endpoint, provider.__build_provider(connection_string, endpoint,
+                credential, key_vault_options)])
 
         selects = kwargs.pop("selects", {SettingSelector("*", "\0")})
 
         provider._trim_prefixes = sorted(kwargs.pop("trimmed_key_prefixes", []), key=len, reverse=True)
 
-        provider._dict = {}
-
         secret_clients = key_vault_options.secret_clients if key_vault_options else {}
 
+        client_count = 0
+
+        # Checking each client for the configuration settings until we find a working one.
+        for client in provider._clients:
+            client_count += 1
+            try:
+                provider.__load_selected_configurations(client, secret_clients, selects, key_vault_options)
+                # If we get here, we have successfully loaded the configuration settings from the client.
+                # We don't need to check the other clients.
+                break
+
+            except ServiceRequestError as e:
+                print("Failed to load configuration settings from " + client[0] + " from Service Request Failure {}."
+                    .format(e))
+                if len(provider._clients) == 1 or len(provider._clients) == client_count:
```

Comment:
Could there be a case where client_count will not be equal to the number of clients when the number of clients is 1?

Reply:
The answer to your first question is No. I am trying to think of why I wrote it like this, but it might have been hedging a bet about an "optional" implementation being added. Though now I am thinking they should be different errors. One is you gave us one client we tried and failed. The other is you gave us a geo replicated store and all failed. @avanigupta thoughts?